CMakeLists.txt

```diff
@@ -162,7 +162,22 @@ if(SIMPATH)
 endif()
 
 # FairRoot only supports ROOT6, so check which version is available
-find_package2(PUBLIC ROOT  VERSION 6.10.00  REQUIRED)
+list(APPEND required_root_components RIO)
+if(BUILD_SIM)
+  list(APPEND required_root_components pythia6 pythia8)
+endif()
+list(APPEND optional_root_components opengl)
+list(APPEND optional_root_components gdml vmc)
+find_package2(PUBLIC ROOT REQUIRED
+  VERSION 6.18.00
+  COMPONENTS ${required_root_components}
+  OPTIONAL_COMPONENTS ${optional_root_components})
+Find_Program(ROOT_CINT_EXECUTABLE
+  NAMES rootcint
+  PATHS ${ROOT_BINDIR}
+  NO_DEFAULT_PATH
+  )
```

Comment:
```suggestion
```

Let's remove this and replace its usage in `cmake/modules/FairRootTargetRootDictionary.cmake` with the imported target, something like:

Reply:
* Changed `FairRootTargetRootDictionary.cmake` as suggested.
* `cmake/scripts/generate_dictionary_root.sh.in` has a `@ROOT_CINT_EXECUTABLE@` in it, and I doubt generator expressions work there. So I think, we need a `find_program`.
* `ROOTMacros.cmake` references this `generate_dictionary_root.sh.in`. So maybe experiments use this stuff? Moved the `find_program` there, so that this works for experiments.